test/factories/smart_proxy_factory.rb

```diff
@@ -21,7 +21,8 @@
 
         smart_proxy_feature = proxy.smart_proxy_features.select { |spf| spf.feature_id == v3_feature.id }.first
         smart_proxy_feature.capabilities = plugins
-        smart_proxy_feature.settings = {pulp_url: "https://#{Socket.gethostname}"}
+        smart_proxy_feature.settings = { pulp_url: "https://#{Socket.gethostname}",
```

Comment:
Unrelated, but typically I like to explicit static values such as `https://pulp.example.com` since this you know that if there is some integration test, it will not magically work sometimes because the host is configured. It also sets the expectation that pulp can be on a remote host and that should be supported.

Reply:
I get the sentiment, but this was pre-existing but I'll leave it for this PR and we can consider removing it outside of this